drivers/SmartThings/zigbee-window-treatment/src/aqara/init.lua

```diff
@@ -1,301 +1,21 @@
-  emit_shade_state_event(device, shadeLevel)
-end
-
-local function window_shade_close_cmd(driver, device, command)
-  send_close_cmd(device, command.component)
-
-  local shadeLevel = getShadeLevelField(device)
-  emit_shade_state_event(device, shadeLevel)
-end
-
-local function window_shade_pause_cmd(driver, device, command)
-  device:send_to_component(command.component, WindowCovering.server.commands.Stop(device))
-
-  local shadeLevel = getShadeLevelField(device)
-  emit_shade_state_event(device, shadeLevel)
-end
-
-local function shade_state_attr_handler(driver, device, value, zb_rx)
-  local state = value.value
-  setShadeStateField(device, state)
-
-  if state == SHADE_STATE_STOP then
-    local shadeLevel = getShadeLevelField(device)
-    emit_shade_state_event(device, shadeLevel)
-
-    local flag = getInitializationField(device)
-    if flag == INIT_STATE_CLOSE then
-      setInitializationField(device, INIT_STATE_REVERSE)
-      send_open_cmd(device, "main")
-    elseif flag == INIT_STATE_OPEN then
-      setInitializationField(device, INIT_STATE_REVERSE)
-      send_close_cmd(device, "main")
-    elseif flag == INIT_STATE_REVERSE then
-      setInitializationField(device, INIT_STATE_DONE)
-      read_pref_attribute(device)
-    end
-  elseif state == SHADE_STATE_OPEN then
-    device:emit_event(capabilities.windowShade.windowShade.opening())
-  elseif state == SHADE_STATE_CLOSE then
-    device:emit_event(capabilities.windowShade.windowShade.closing())
-  end
-end
-
-local function current_position_attr_handler(driver, device, value, zb_rx)
-  local level = value.value
-  if level > 100 then
-    level = 100
-  end
-  level = utils.round(level)
-
-  setShadeLevelField(device, level)
-  emit_shade_level_event(device, level)
-
-  local shadeState = getShadeStateField(device)
-  if shadeState == SHADE_STATE_STOP then
-    emit_shade_state_event(device, level)
-  end
-end
-
-local function pref_attr_handler(driver, device, value, zb_rx)
-  local initialized = string.byte(value.value, 3) & 0xFF
-  local flag = getInitializationField(device)
-  if flag == INIT_STATE_DONE then
-    device:emit_event(initialized == 1 and deviceInitialization.initializedState.initialized() or
-      deviceInitialization.initializedState.notInitialized())
-  end
-end
-
 local aqara_window_treatment_handler = {
   NAME = "Aqara Window Treatment Handler",
-  lifecycle_handlers = {
-    added = device_added,
-    doConfigure = do_configure,
-    infoChanged = device_info_changed
-  },
-  capability_handlers = {
-    [capabilities.windowShadeLevel.ID] = {
-      [capabilities.windowShadeLevel.commands.setShadeLevel.NAME] = window_shade_level_cmd
-    },
-    [capabilities.windowShade.ID] = {
-      [capabilities.windowShade.commands.open.NAME] = window_shade_open_cmd,
-      [capabilities.windowShade.commands.close.NAME] = window_shade_close_cmd,
-      [capabilities.windowShade.commands.pause.NAME] = window_shade_pause_cmd
-    },

```

Comment:
as far as I can tell, both sub-drivers _at least_ still have the same open, close, and pause commands. They can inherit them from the parent driver rather than making this driver completely empty.

Reply:
`refresh` is also the same for both

Reply:
`shade_level_report_handler` also seems to be the same for both.

Reply:
You should re-examine what functions of these two sub-drivers are _actually_ distinct. Everything in common should be here in the base driver.

Reply:
Where can I find a example code to inherit sub-driver from the parent?

Reply:
Include `sub_drivers = {...}` in your sub-driver. Any handler in the sub-driver that isn't overwritten by the child will default to the parent's.

see, for example, the `axis` driver: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/tree/main/drivers/SmartThings/zigbee-window-treatment/src/axis

or your own motion sensor driver:https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/tree/main/drivers/SmartThings/zigbee-motion-sensor/src/aqara

Reply:
I updated the code to override the child's handler according to your guide.